qrand/platforms/qsharp/job.py

```diff
@@ -31,34 +35,129 @@
+        self._backend: QsharpBackend = backend
 
     @property
     def circuit(self) -> QsharpCircuit:
-        raise NotImplementedError(self.ERROR_MSG)
+        return self._circuit
 
     @circuit.setter
     def circuit(self, circuit: QsharpCircuit) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        if self.backend.max_qubits < circuit.num_qubits:
+            raise RuntimeError(
+                f"Failed to assign QsharpCircuit for QsharpJob. Number of \
+                qubits in QsharpCircuit unsupported by this job's Backend: \
+                {self.backend.max_qubits}<{circuit.num_qubits}."
+            )
+        self._circuit: QsharpCircuit = circuit
 
     @property
     def num_measurements(self) -> int:
-        raise NotImplementedError(self.ERROR_MSG)
+        return self._shots * self._experiments
 
     @num_measurements.setter
     def num_measurements(self, num_measurements: Optional[int]) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        num_measurements = (
+            num_measurements
+            if num_measurements
+            and type(num_measurements) is int
+            and 0 < num_measurements
+            else self.backend.max_measurements
+        )
+        if self.backend.max_measurements < num_measurements:
+            warn(
+                f"Number of measurements unsupported by the job's Backend: \
+                {self.backend.max_measurements}<{num_measurements}. \
+                Using max_measurements instead.",
+                UserWarning,
+            )
+            num_measurements = self.backend.max_measurements
+        self._shots, self._experiments = compute_bounded_factorization(
+            num_measurements,
+            self.backend.max_shots,
+            self.backend.max_experiments,
+        )
+
+    def _generate_code(self) -> QSharpCallable:
+        qsharp_code = """
+        open Microsoft.Quantum.Intrinsic;
+        open Microsoft.Quantum.Measurement;
+        open Microsoft.Quantum.Arrays;
+
+        operation Program():String[]{{
+
+        use q = Qubit[{num_qubits}];
+        mutable value = ConstantArray({num_qubits},"0");
+
+        {gates}
+
+        ResetAll(q);
+        return value;
+        }}
+        """
+        return compile(
+            qsharp_code.format(
+                num_qubits=self.circuit.num_qubits, gates=self.circuit.gates
+            )
+        )
 
     def execute(self) -> List[str]:
-        raise NotImplementedError(self.ERROR_MSG)
+        self.program = self._generate_code()
+        result: List[str] = []
+
+        if self.backend.resource_id is None or self.backend.target_id is None:
+            result = self.program.simulate()
+        else:
+            azure.connect(resourceId=self.backend.resource_id)
+            azure.target(targetId=self.backend.target_id)
+            result = azure.execute(
+                self.program,
+                shots=self.num_measurements,
```

Comment:
I'll resort to `num_measurements`.

Reply:
However, can you pass this same thing to simulate? If you can't then it should be removed (set to 1), since it would make booth behaviors inconsistent. I believe you already took care of shots by means of a for loop, so let me know what you decide here :-)

Reply:
Set azure `shots` to one. Behavior is consistent with `simulate`